src/prefect/cli/deploy.py

```diff
@@ -726,10 +727,160 @@ def _merge_with_default_deploy_config(deploy_config: Dict):
+        )
+    if not confirm(
+        f"Is [green]{branch}[/] the correct branch to pull your flow code from?",
+        default=True,
+        console=console,
+    ):
+        branch = prompt(
+            "Please enter the branch to pull your flow code from",
+            default="main",
+            console=console,
+        )
+    token_secret_block_name = None
+    if confirm("Is this a private repository?", console=console):
+        token_secret_block_name = f"deployment-{slugify(deploy_config['name'])}-{slugify(deploy_config['flow_name'])}-repo-token"
+        create_new_block = False
+        prompt_message = (
+            "Please enter a token that can be used to access your private"
+            " repository. This token will be saved as a secret via the Prefect API"
+        )
+
+        try:
+            await Secret.load(token_secret_block_name)
+            if not confirm(
+                (
+                    "We found an existing token saved for this deployment. Would"
+                    " you like use the existing token?"
+                ),
+                default=True,
+                console=console,
+            ):
+                prompt_message = (
+                    "Please enter a token that can be used to access your private"
+                    " repository (this will overwrite the existing token saved via"
+                    " the Prefect API)."
+                )
+
+                create_new_block = True
+        except ValueError:
+            create_new_block = True
+
+        if create_new_block:
+            try:
+                repo_token = prompt(
+                    prompt_message,
+                    console=console,
+                    password=True,
+                )
+            except GetPassWarning:
+                # Handling for when password masking is not supported
+                repo_token = prompt(
+                    prompt_message,
+                    console=console,
+                )
+            await Secret(
+                value=repo_token,
+            ).save(name=token_secret_block_name, overwrite=True)
+
+    git_clone_step = {
+        "prefect.deployments.steps.git_clone": {
+            "repository": remote_url,
+            "branch": branch,
+        }
+    }
+
+    if token_secret_block_name:
+        git_clone_step["prefect.deployments.steps.git_clone"]["access_token"] = (
+            "{{ prefect.blocks.secret." + token_secret_block_name + " }}"
+        )
+
+    return [git_clone_step]
+
+
+async def _generate_pull_step_for_build_docker_image(
+    console: Console, deploy_config: Dict, auto: bool = True
+):
+    pull_step = {}
+    if auto:
+        pull_step["path"] = "/opt/prefect/{{ name }}"
+    else:
+        pull_step["path"] = prompt(
+            "What is the path to your flow code in your Dockerfile?",
+            default="/opt/prefect/{{ name }}",
+            console=console,
+        )
```

Comment:
The `{{ name }}` placeholder won't template correctly in most cases, so using the current directory name is safer. It aligns with the behavior in `prefect-docker` as well.

```suggestion
    dir_name = os.path.basename(os.getcwd())
    if auto:
        pull_step["directory"] = f"/opt/prefect/{dir_name}"
    else:
        pull_step["directory"] = prompt(
            "What is the path to your flow code in your Dockerfile?",
            default=f"/opt/prefect/{dir_name}",
            console=console,
        )
```